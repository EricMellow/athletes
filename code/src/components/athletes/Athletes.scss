.trainheroic-code-test {
    .profile-picture {
      border-radius: 100px;
      height: 150px;
    }

    .athlete-container {
      display: flex;
      flex-wrap: wrap;
      justify-content: space-evenly;
      width: 60%;
      margin: auto;
    }

    .athlete-card {
      min-width: 250px;
      text-align: center;
      padding: 20px;
      margin: 20px;
    }

    @media only screen and (max-width: 1100px) {
      .athlete-container {
        width: 80%;
      }
    }

    @media only screen and (max-width: 825px) {
      .athlete-container {
        width: 100%;
      }
    }
}