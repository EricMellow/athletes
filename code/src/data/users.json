[
    {
        "id": 5101,
        "nameFirst": "Abby",
        "nameLast": "Smith",
        "profilePhotoUrl": "http://s3-us-west-2.amazonaws.com/trainheroic-prod/profile/1509665304_thumb.jpg",
        "slug": "abby-smith"
    },
    {
        "id": 9705,
        "nameFirst": "Dave",
        "nameLast": "Jones",
        "profilePhotoUrl": "http://trainheroic-prod.s3.us-west-2.amazonaws.com/profile/1589716488_orig.jpg",
        "slug": "dave-jones"
    },
    {
        "id": 15321,
        "nameFirst": "Becky",
        "nameLast": "Davis",
        "profilePhotoUrl": "http://trainheroic-prod.s3.us-west-2.amazonaws.com/profile/1597425796_orig.jpg",
        "slug": "becky-davis"
    },
    {
        "id": 22677,
        "nameFirst": "Barry",
        "nameLast": "Moore",
        "profilePhotoUrl": "http://trainheroic-prod.s3.us-west-2.amazonaws.com/profile/1561656476_orig.jpg",
        "slug": "barry-moore"
    },
    {
        "id": 29891,
        "nameFirst": "Jamie",
        "nameLast": "Johnston",
        "profilePhotoUrl": "http://trainheroic-prod.s3.us-west-2.amazonaws.com/profile/1581826381_orig.jpg",
        "slug": "jamie-johnston"
    }
]