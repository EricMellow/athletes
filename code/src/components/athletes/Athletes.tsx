import "./Athletes.scss"

type AthletesProps = { athletes: Array<TrainHeroic.User> };

export default function Athletes({ athletes }: AthletesProps) {
  let athleteCards: Array<object> = [];
  if (athletes.length) {
    athleteCards = athletes.map(athlete => {
      return (
        <div key={athlete.id} className="athlete-card">
          <img src={athlete.profilePhotoUrl} alt="Athlete profile" className="profile-picture"></img>
          <h3>{athlete.nameFirst} {athlete.nameLast}</h3>
        </div>
      );
    });
  }

  return (
    <div className="athlete-container">
      {athleteCards.length === 0 && <div className="no-athletes">
        We're sorry, but you don't currently have any athletes.
      </div>}
      {athleteCards.length >= 1 && athleteCards}
    </div>
  );
}