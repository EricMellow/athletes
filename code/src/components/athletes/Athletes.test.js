import React from 'react';
import Athletes from './Athletes';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import users from '../../data/users.json';

Enzyme.configure({ adapter: new Adapter() });

describe('Athletes', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      athletes: users
    };
    wrapper = shallow(<Athletes {...props}/>);
  });

  it('should match the snapshot with athlete props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot without athlete props', () => {
    const props = {
      athletes: []
    };
    wrapper = shallow(<Athletes {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  

});
