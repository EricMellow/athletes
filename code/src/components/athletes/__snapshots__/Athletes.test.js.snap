// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Athletes should match the snapshot with athlete props 1`] = `ShallowWrapper {}`;

exports[`Athletes should match the snapshot without athlete props 1`] = `ShallowWrapper {}`;
